tests/integration/db/taxi_data_utils.py

```diff
@@ -95,119 +92,74 @@ def _execute_taxi_partitioning_test_cases(
 
         column_name: str = taxi_partitioning_test_cases.test_column_name
         column_names: List[str] = taxi_partitioning_test_cases.test_column_names
 
         # 2. Set partitioner in DataConnector config
-        data_connector_config: dict = {
-            "class_name": "ConfiguredAssetSqlDataConnector",
-            "assets": {
-                data_asset_name: {
-                    "partitioner_method": test_case.partitioner_method_name,
-                    "partitioner_kwargs": test_case.partitioner_kwargs,
-                }
-            },
-        }
-
-        # noinspection PyTypeChecker
-        context.add_datasource(
-            name=datasource_name,
-            class_name="Datasource",
-            execution_engine={
-                "class_name": "SqlAlchemyExecutionEngine",
-                "connection_string": connection_string,
-            },
-            data_connectors={data_connector_name: data_connector_config},
-        )
-
-        datasource: BaseDatasource = context.get_datasource(datasource_name=datasource_name)
-
-        data_connector: ConfiguredAssetSqlDataConnector = datasource.data_connectors[
-            data_connector_name
-        ]
 
-        # 3. Check if resulting batches are as expected
-        # using data_connector.get_batch_definition_list_from_batch_request()
-        batch_request: BatchRequest = BatchRequest(
-            datasource_name=datasource_name,
-            data_connector_name=data_connector_name,
-            data_asset_name=data_asset_name,
+        # NOTE: this getattr with method names pattern is horrible; consider refactoring
+        add_datasource_method = getattr(context.sources, add_datasource_method_name)
+        datasource: SQLDatasource = add_datasource_method(
+            datasource_name, connection_string=connection_string
+        )
+        asset = datasource.add_table_asset(data_asset_name, table_name=table_name)
+        add_batch_definition_method = getattr(
+            asset, test_case.add_batch_definition_method_name or "MAKE THIS REQUIRED"
         )
-        batch_definition_list: List[LegacyBatchDefinition] = (
-            data_connector.get_batch_definition_list_from_batch_request(batch_request=batch_request)
+        batch_definition: BatchDefinition = add_batch_definition_method(
+            name=batch_definition_name, **test_case.add_batch_definition_kwargs
         )
-        print(len(batch_definition_list), "batch definitions found")
+
+        # 3. Check if resulting batches are as expected
+        batch_request = batch_definition.build_batch_request()
+        batch_list = asset.get_batch_list_from_batch_request(batch_request)
+        print(len(batch_list), "batch definitions found")
         print(test_case.num_expected_batch_definitions, "expected batch definitions")
-        assert len(batch_definition_list) == test_case.num_expected_batch_definitions
+        assert len(batch_list) == test_case.num_expected_batch_definitions
```

Comment:
nit: I would move the prints to be the second argument to the assert:
```assert len(...) == ..., f"Found {len(batch_list)} batch definitions but expected {test_case.num_expected_batch_definitions}"```
Then this will appear in the assert error.